net/strongswan/files/swanctl.init

```diff
@@ -148,6 +152,88 @@ is_aead() {
 	return 1
 }
 
+authority_one_exclusive() {
+	fatal "Configure either cacert, file, or handle per authority section!"
+}
+
+config_authority() {
+	local config_name="$1"
+
+	local enabled
+
+	local cacert
+	local cert_uri_base
+	local crl_uris
+	local file
+	local handle
+	local module
+	local oscp_uris
+	local slot
+
+	config_get_bool enabled "$1" enabled 0
+	[ $enabled -eq 0 ] && return
+
+	config_get cacert "$1" cacert ""
+	config_get cert_uri_base "$1" cert_uri_base ""
+	config_get crl_uris "$1" crl_uris ""
+	config_get file "$1" file ""
+	config_get handle "$1" handle ""
+	config_get module "$1" module ""
+	config_get oscp_uris "$1" oscp_uris ""
+	config_get slot "$1" slot ""
+
+	[ -z "$cacert" ] && [ -z "$file" ] && [ -z "$handle" ] && authority_one_exclusive
+
+	[ -n "$cacert" ] && [ -n "$file" ] && authority_one_exclusive
+	[ -n "$cacert" ] && [ -n "$handle" ] && authority_one_exclusive
+	[ -n "$file" ] && [ -n "$handle" ] && authority_one_exclusive
+
+	swanctl_xappend0 "authorities {"
+	swanctl_xappend1 "$config_name {"
+	[ -n "$cacert" ] && swanctl_xappend2 "cacert = $cacert"
+	[ -n "$cert_uri_base" ] && swanctl_xappend2 "cert_uri_base = $cert_uri_base"
+
+	local quoted_comma_list=""
+	config_list_foreach "$1" crl_uris append_quoted_comma_list
+	[ -n "$quoted_comma_list" ] && swanctl_xappend3 "crl_uris = $quoted_comma_list"
+
+	[ -n "$file" ] && swanctl_xappend2 "file = $file"
+	[ -n "$handle" ] && swanctl_xappend2 "handle = $handle"
+	[ -n "$module" ] && swanctl_xappend2 "module = $module"
+
+	local quoted_comma_list=""
+	config_list_foreach "$1" oscp_uris append_quoted_comma_list
+	[ -n "$quoted_comma_list" ] && swanctl_xappend3 "oscp_uris = $quoted_comma_list"
+
+	[ -n "$slot" ] && swanctl_xappend2 "slot = $slot"
+	swanctl_xappend1 "}"
+	swanctl_xappend0 "}"
+
+	swanctl_xappend0 ""
+}
+
+add_ah_proposal() {
```

Comment:
Who uses AH (besides Microsoft) and **why**?  ESP provides you authenticity, integrity, *and* confidentiality.

Reply:
Having AH support is just extra sugar now. In the last 8 years I have only seen three configs with AH.